### PR TITLE
guard against bad hinge-constraint settings

### PR DESCRIPTION
**Pull request by sethalves**
- check hinge parameters for zero-length axis parameters.

### CODE REVIEW

File: libraries/physics/src/ObjectConstraintHinge.cpp [hunk 1]
```diff
@@ -104,12 +104,27 @@ btTypedConstraint* ObjectConstraintHinge::getConstraint() {
         return nullptr;
     }
 
+    if (glm::length(axisInA) == 0.0f) {
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectConstraintHinge.cpp]: Check for length < EPSILON.  Very very very small numbers can also cause problems here.

File: libraries/physics/src/ObjectConstraintHinge.cpp [hunk 2]
```diff
@@ -104,12 +104,27 @@ btTypedConstraint* ObjectConstraintHinge::getConstraint() {
...
+        qCWarning(physics) << "hinge axis cannot be a zero vector";
+        axisInA = DEFAULT_HINGE_AXIS;
+    }
+
+    axisInA = glm::normalize(axisInA);
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectConstraintHinge.cpp]: Put this in the `else` statement.

File: libraries/physics/src/ObjectConstraintHinge.cpp [hunk 3]
```diff
@@ -104,12 +104,27 @@ btTypedConstraint* ObjectConstraintHinge::getConstraint() {
...
         if (!rigidBodyB) {
             return nullptr;
         }
+
+        if (glm::length(axisInB)) {
+            qCWarning(physics) << "hinge axis cannot be a zero vector";
+            axisInB = DEFAULT_HINGE_AXIS;
+        }
+
+        axisInB = glm::normalize(axisInB);
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectConstraintHinge.cpp]: Similarly here.  Compare with EPSILON and put the normalization in the `else`.

**Comment by sethalves** [comment]
to test:
- paste into console:
```javascript
var carBody = Entities.addEntity({ type: "Sphere",
                               position: Vec3.sum(MyAvatar.position, {x: 2.0, y:0, z:0}),
                               dimensions: { x: 5, y: 1, z: 2 },
                               color: { red: 0, green: 0, blue: 0 },
                               dynamic: true,
                               gravity: {y:-9.8},
                               });
Entities.addAction("hinge", wheelFrontRight, { otherEntityID: carBody } );
```

**Comment by AndrewMeadows** [comment]
I cloned this branch locally, built, and tested this: it fixes the crash mode.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c693687927eba1482e7f21e96523e346c0cd4787 available at: https://gist.github.com/cefa9813b1d7db487779b28a7cddea61<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10376-c693687.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10376-c693687.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f867a8e7184e66c5b22b893f8bab49a027217179 available at: https://gist.github.com/cefa9813b1d7db487779b28a7cddea61<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10376-f867a8e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10376-f867a8e.dmg

**Pull request merged**